lib/ide/src/controller/text.rs

```diff
@@ -88,18 +89,36 @@ impl Handle {
     }
 
     /// Read file's content.
-    pub fn read_content(&self) -> impl Future<Output=Result<String,RpcError>> {
+    pub async fn read_content(&self) -> Result<String,RpcError> {
+        use FileHandle::*;
         match self.file_handle() {
-            FileHandle::PlainText {path,mut file_manager} => file_manager.read(path),
-            FileHandle::Module {..}                       => todo!(),
+            PlainText {path,mut file_manager} => file_manager.read(path).await,
+            Module    {controller}            => Ok(controller.code())
         }
     }
 
     /// Store the given content to file.
-    pub fn store_content(&self, content:String) -> impl Future<Output=Result<(),RpcError>> {
+    pub async fn store_content(&self, content:String) -> FallibleResult<()> {
```

Comment:
Do we want to change this?
I believe that this keeps self borrowed for the whole duration of the call (that after obtaining handle to file manager before waiting, doesn't need it anymore)

Previously I had trouble with your version of the signature here,

Reply:
You didn't end a sentence, did you?

And you are wrong. The only borrow in inside `self.file_handle()` call. Once that function returns a cloned_ref file handle, borrow is no more